runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/PCollectionsTest.java

```diff
@@ -154,7 +155,15 @@ public Instant maxTimestamp() {
 
     @Override
     public boolean isCompatible(WindowFn<?, ?> other) {
-      return other != null && this.getClass().equals(other.getClass());
+      return other instanceof CustomWindows;
```

Comment:
This is OK because it is a private test class, but usually a subclass could even be incompatible.